COPYING

```diff
@@ -1,118 +1,41 @@
-duplicate all of the original copyright notices and associated
-disclaimers.
-
-2.  You may apply bug fixes, portability fixes and other modifications
-derived from the Public Domain or from the Copyright Holder.  A Package
-modified in such a way shall still be considered the Standard Version.
-
-3.  You may otherwise modify your copy of this Package in any way,
-provided that you insert a prominent notice in each changed file stating
-how and when AND WHY you changed that file, and provided that you do at
-least ONE of the following:
-
-a) place your modifications in the Public Domain or otherwise make them
-Freely Available, such as by posting said modifications to Usenet or an
-equivalent medium, or placing the modifications on a major archive site
-such as uunet.uu.net, or by allowing the Copyright Holder to include
-your modifications in the Standard Version of the Package.
-
-b) use the modified Package only within your corporation or organization.
-
-c) rename any non-standard executables so the names do not conflict with
-standard executables, which must also be provided, and provide separate
-documentation for each non-standard executable that clearly documents
-how it differs from the Standard Version.
-
-d) make other distribution arrangements with the Copyright Holder.
-
-4.  You may distribute the programs of this Package in object code or
-executable form, provided that you do at least ONE of the following:
-
-a) distribute a Standard Version of the executables and library files,
-together with instructions (in the manual page or equivalent) on where
-to get the Standard Version.
-
-b) accompany the distribution with the machine-readable source of the
-Package with your modifications.
-
-c) accompany any non-standard executables with their corresponding
-Standard Version executables, giving the non-standard executables
-non-standard names, and clearly documenting the differences in manual
-pages (or equivalent), together with instructions on where to get the
-Standard Version.
-
-d) make other distribution arrangements with the Copyright Holder.
-
-5.  You may charge a reasonable copying fee for any distribution of this
-Package.  You may charge any fee you choose for support of this Package.
-YOU MAY NOT CHARGE A FEE FOR THIS PACKAGE ITSELF.  However, you may
-distribute this Package in aggregate with other (possibly commercial)
-programs as part of a larger (possibly commercial) software distribution
-provided that YOU DO NOT ADVERTISE this package as a product of your
-own.
-
-6.  The name of the Copyright Holder may not be used to endorse or
-promote products derived from this software without specific prior
-written permission.
-
-7.  THIS PACKAGE IS PROVIDED "AS IS" AND WITHOUT ANY EXPRESS OR IMPLIED
-WARRANTIES, INCLUDING, WITHOUT LIMITATION, THE IMPLIED WARRANTIES OF
-MERCHANTIBILITY AND FITNESS FOR A PARTICULAR PURPOSE.
-
-				The End
+SPDX-License-Identifier: BSD-3-Clause
```

Comment:
I think BSD-3-Clause is not a valid license for the whole project. I think there is none, since `contrib/atudel` is BSD-4-Clause which is not compatible with GPL https://en.wikipedia.org/wiki/BSD_licenses#4-clause_license_(original_%22BSD_License%22), yet `src/vipw.c` is licensed under GPL.
I suggest dropping `contrib/atudel` or splitting it out to a separate project.

If `vipw.c` stays in the source under GPL, then the the license of the project should be GPL (with BSD parts).

Reply:
I don't think so since those are all separate programs, and this situation is far from being unique; I've seen quite a few projects with non-GPL parts, e.g. an application and a client library to talk to it.

Reply:
@andrewshadura Do you have some links? I guess you question the GPL as the whole project's license and not the problem with `atudel`.

Reply:
Most definitely GPL is not the project’s license, but just the license of one independent program.

Reply:
@andrewshadura I agree that the project is belived to be BSD licensed, but looking at the current state the release tarballs can't be distributed under BSD. If we remove `atudel` the resulting tarball can be distributed under GPL, but not under BSD (3 clause), because it contains a GPL licensed source file. If we cut `vipw.c`, the remaining tarball can be distributed under BSD (3 clause).
From Debian's perspective I'm OK with keeping `vipw.c` in the tarball, but I have to exclude `atudel`, for sure, to keep `shadow`'s source in `main`.
It could be claimed, that `atudel` and `vipw.c` are only bundled, but this needs an explanation in the COPYING file, IMO.

Reply:
@rbalint, you’re misinterpreting it. `atudel` has no relationship whatsoever to `vipw` except being shipped in the same tarball. It is not a derivative work of `vipw`, it’s not based on it, linked with it or anything really except being bundled together.

You cannot claim that the license of the whole package is GPL, because it’s BSD-3-Clause for most files, GPL for `vipw` and BSD-4-Clause for `atudel`. I don’t think a concept of a "license for the whole package" makes sense at all in this case.

Reply:
Originally I'd planned to remove the copyright statement from the files which fall under the 3 clause BSD license.  The COPYING file then just listed the default.  It explicitly says that files may list alternative licenses, which vipw does.

Perhaps I should add a paragraph to say that all new files are to be BSD-3.  But as @andrewshadura says, vipw and atudel are the exceptions.  It may make a difference from a debian packaging perspective, but for the git repo all I'm doing is documenting the existing licenses.  We will not relicense other files just because vipw is under src/.  In particular, section 2 of the GPL says:

```
In addition, mere aggregation of another work not based on the Program with the Program (or with a work based on the Program) on a volume of a storage or distribution medium does not bring the other work under the scope of this License. 
```

The git repo is (in my estimation) a storage or distribution medium.  No other programs in shadow are based on vipw.